core/io/marshalls.cpp

```diff
@@ -1152,6 +1237,22 @@ Error encode_variant(const Variant &p_variant, uint8_t *r_buffer, int &r_len, bo
 				flags |= ENCODE_FLAG_OBJECT_AS_ID;
 			}
 		} break;
+		case Variant::ARRAY: {
+			Array array = p_variant;
+			if (array.is_typed()) {
+				Ref<Script> script = array.get_typed_script();
+				if (script.is_valid()) {
+					ERR_FAIL_COND_V(!p_full_objects, ERR_UNAVAILABLE);
+					flags |= ENCODE_FIELD_TYPED_ARRAY_SCRIPT;
```

Comment:
It looks a little weird (flags vs field).

Reply:
Indeed. Maybe `flags` warrants a rename to `encode_data` or the like.

Reply:
I used `header` as in [documentation](https://docs.godotengine.org/en/stable/tutorials/io/binary_serialization_api.html#packet-specification):

> The lowest value two bytes are used to determine the type, while the highest value two bytes contain flags

Reply:
Sorry for going on with the bikeshedding, but my point is to keep flags named `*_FLAG_*` (so it's very easy to understand they are independent bits) and the new values `*_FIELD_*` (so it's easy to understand it's a bunch of bits representing some other type). Then, both _flags_ and _fields_ would be considered part of the "header," "encoding data", or any other suiting name.

Reply:
Done.

```mermaid
flowchart LR
    header-->type
    header-->data
    data-->flag
    data-->field
```